CONTRIBUTING.md

```diff
@@ -0,0 +1,114 @@
+# How to Contribute
+## Creating an issue
+
+Raising an issue is the easiest and most effective way to begin engaging with BlackParrot (or any open-source project).  Being thorough and detailed in the issue makes it easier to communicate, requiring fewer roundtrip messages (increasing time to resolution).
+
+Generally, there are three types of issues raised in BlackParrot.
+
+### Bug reports
+**Something doesn't look right to me.**
+- Suggested fields:
+    - Descriptive title.
+    - Module/tool/makefile/script in question.
+    - Current behavior (steps for reproduction).
+    - Expected behavior.
+    - Proposed solution(s), including scope of work and affected modules.
+    - Methods of verification.
+- Example: Power waste by propagating data when stalled.
```

Comment:
Example doesn't seem to be "complete", in that we describe a bunch of things the issue should specify, but our example is a single sentence.

Reply:
Added complete examples